src/facade/arg_parser.h

```diff
@@ -0,0 +1,176 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/strings/match.h>
+
+#include <string_view>
+
+#include "facade/error.h"
+#include "facade/facade_types.h"
+
+namespace facade {
+
+struct ArgumentParser {
+  struct NextProxy;
+  friend struct NextProxy;
```

Comment:
Do they still need to be friends if they're inside the class?

Reply:
True, I forgot they're by default